python/hail/api2/matrixtable.py

```diff
@@ -2222,5 +2222,118 @@ def unpersist(self):
                       tolerance=numeric)
     def _same(self, other, tolerance=1e-6):
         return self._jvds.same(other._jvds, tolerance)
-        
+
+    @handle_py4j
+    @typecheck(datasets=matrix_table_type)
+    def union_rows(*datasets):
+        """Take the union of dataset rows.
+
+        Examples
+        --------
+
+        .. testsetup::
+
+            dataset_to_union = dataset
+            dataset_to_union_2 = dataset
+
+        Union the rows of two datasets:
+
+        >>> dataset_result = dataset.union_rows(dataset_to_union)
+
+        Given a list of datasets, take the union of all rows:
+
+        >>> all_datasets = [dataset, dataset_to_union, dataset_to_union_2]
+
+        The following three syntaxes are equivalent:
+
+        >>> dataset_result = dataset.union_rows(dataset_to_union, dataset_to_union_2)
+        >>> dataset_result = all_datasets[0].union_rows(*all_datasets[1:])
+        >>> dataset_result = MatrixTable.union_rows(*all_datsets)
+
+        Notes
+        -----
+
+        In order to combine two datasets, four requirements must be met:
+
+         - The column keys must be identical (in both type and equality, field
+           order within structs matters).
+         - The row key schema and row field schemas must match.
+         - The entry schemas must match.
+
+        The column fields in the resulting dataset are simply the column
```

Comment:
This line is confusing to me -- probably because I am not used to the new naming scheme. I think it would be clearer if it was "The column annotations in the resulting dataset are simply the column annotations from the first dataset; the column annotation schemas do not need to match".

Reply:
replaced the "annotation" with fields for consistency.